### PR TITLE
Add getExecOutput function

### LINKED ISSUE #769: Easier to get command output
**Describe the enhancement**

There are a few Github Actions that require the command output, like [docker/build-push-action](https://github.com/docker/build-push-action/blob/master/src/exec.ts), [bjerkio/kopier](https://github.com/bjerkio/kopier/blob/9ec42c990dfd6ee394a4b16d0be2b054259552d3/src/git-commit.ts#L25-L47) and [pulumi/action](https://github.com/pulumi/actions/blob/master/src/libs/exec.ts).

I've personally added this functionality twice so far, so I thought it might be something we should consider adding to `@actions/exec` directly?

**Code Snippet**

```typescript
import { getCommandOutput } from '@actions/exec';

const output = await getCommandOutput('echo', ['hello', 'world']);

// output = "hello world"
```

### PR DESCRIPTION
**Pull request by luketomlinson**
Adds a convenience function go more easily get the output of an exec command. Rather than having to explicitly define listeners in `ExecOptions` this method will return the collected output of `stdOut` and `stdErr` in addition to the exit code. This should make usage a little less awkward in situations where users need the output of an `exec` command. 

Resolves #769

### CODE REVIEW

File: packages/exec/__tests__/scripts/stdoutoutputlarge.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+//Default highWaterMark for readable stream buffers us 64K (2^16)
+//so we go over that to get more than a buffer's worth
+process.stdout.write('a'.repeat(2**24));
```
Comment (thboop) [inline packages/exec/__tests__/scripts/stdoutoutputlarge.js:3]: Nit: can we add tests that include newlines and stderr in the output.

File: packages/exec/src/exec.ts [hunk 2]
```diff
@@ -28,3 +28,44 @@ export async function exec(
...
+
+  const listeners: ExecListeners = {
+    ...options?.listeners,
+    stdout: stdOutListener,
+    stderr: stdErrListener
+  }
+
+  const exitCode = await exec(commandLine, args, {...options, listeners})
+
+  //return undefined for stdout/stderr if they are empty
```
Comment (thboop) [inline packages/exec/src/exec.ts:65]: Why undefined over the empty string? If there's no output, returning "" should be fine

File: packages/exec/__tests__/scripts/stdoutoutputlarge.js [hunk 3]
```diff
@@ -0,0 +1,3 @@
+//Default highWaterMark for readable stream buffers us 64K (2^16)
```
Comment (thboop) [inline packages/exec/__tests__/scripts/stdoutoutputlarge.js:1]: Nit: Might be worth adding scripts to test output with newlines as well

File: packages/io/src/io.ts [hunk 4]
```diff
@@ -17,7 +17,6 @@ export interface CopyOptions {
   /** Optional. Whether to copy the source directory along with all the files. Only takes effect when recursive=true and copying a directory. Default is true*/
   copySourceDirectory?: boolean
 }
-
```
Comment (thboop) [inline packages/io/src/io.ts:20]: Revert this change please!

**Review by thboop — DISMISSED** [review]
LGTM minor thoughts

**Review by thboop — COMMENTED** [review]
Tests appear to be failing

File: packages/io/src/io.ts (hunk 4 above)
Comment (luketomlinson) [reply packages/io/src/io.ts:20]: will do. I believe it was just linting

**Review by thboop — APPROVED** [review]
LGTM

**Comment by t1m0thyj** [comment]
The functionality introduced in this PR is very useful. For now I've copied the `getExecOutput` function into my GH action. Any idea when an update to `@actions/exec` will be published that includes this change? Thanks!

**Comment by luketomlinson** [comment]
@t1m0thyj This should go out next week. See [this PR](https://github.com/actions/toolkit/pull/834)
